kedro/config/omegaconf_config.py

```diff
@@ -286,7 +286,13 @@ def load_and_merge_dir_config(  # pylint: disable=too-many-arguments
             return OmegaConf.to_container(
                 OmegaConf.merge(*aggregate_config, self.runtime_params), resolve=True
             )
-        return OmegaConf.to_container(OmegaConf.merge(*aggregate_config), resolve=True)
+        return {
+            k: v
+            for k, v in OmegaConf.to_container(
+                OmegaConf.merge(*aggregate_config), resolve=True
+            ).items()
+            if not k.startswith("_")
+        }
```

Comment:
Just want to point out that this doesn't affect `catalog` only, it is more generalised to anything other than `parameters`, for example `sparks`.

Reply:
Very good point!! I'll clarify that in the docs.

Reply:
I guess it's a bit more nuanced. Templating in other files don't need the `_`, that's specifically needed for the catalog validation, but it's true that the `_` values in any files will be ignored from duplication checks etc.